packages/cosmo/package.py

```diff
@@ -205,11 +207,24 @@ def test(self):
                     env['REAL_TYPE'] = 'FLOAT'
                 if '~cppdycore' in self.spec:
                     env['JENKINS_NO_DYCORE'] = 'ON'
-                run_testsuite = Executable('sbatch submit.' + self.spec.variants['slave'].value + '.slurm')
-                run_testsuite()
+                if self.spec.variants['slave'].value == 'tsa_rh7.7':
+                    run_testsuite = 'sbatch -W --reservation=rh77 submit.tsa.slurm'
+                else:
+                    run_testsuite = 'sbatch -W submit.' + self.spec.variants['slave'].value + '.slurm'
+                os.system(run_testsuite)
+                cat_testsuite = 'cat testsuite.out'
+                os.system(cat_testsuite)
+                check_testsuite = './testfail.sh'
+                os.system(check_testsuite)
         if '+serialize' in self.spec:
             with working_dir(prefix.cosmo + '/ACC'):
-                get_serialization_data = Executable('./test/serialize/generateUnittestData.py -v -e cosmo_serialize --mpirun=srun')
-                get_serialization_data()
+                if self.spec.variants['slave'].value == 'tsa_rh7.7':
+                    get_serialization_data = './test/serialize/generateUnittestData.py -v -e cosmo_serialize --mpirun=\'srun --reservation=rh77\' >> serialize_log.txt; grep \'Generation failed\' serialize_log.txt | wc -l'
+                else:
+                    get_serialization_data = './test/serialize/generateUnittestData.py -v -e cosmo_serialize --mpirun=srun >> serialize_log.txt; grep \'Generation failed\' serialize_log.txt | wc -l'
```

Comment:
I think it would be better to do the regex in python, rather than redirecting into a file and doing the regex all in one single bash command. 
something like 
```
output = subprocess.run('./test/serialize/generateUnittestData.py -v -e cosmo_serialize --mpirun=\'srun --reservation=rh77\', capture_output=True)
if re.match(output.stdout, "Generation failed"):
    raise ...
```
However I also think we should do that with python3, not  python2. So if this work I would suggest to go ahead with this and we can change it later if you agree.